vars.tf

```diff
@@ -2,9 +2,16 @@ variable "env" {
   type        = string
   description = "Environment (Typically one of 'test', 'stage' or 'prod')"
 }
-variable "lambda_code_dir" {}
-variable "lambda_name" {}
-variable "lambda_source_dir_name" {
+variable "lambda_dir" {
+  type = string
+  description = "Common dir for lambda code"
+  default = "../../lambda"
+}
+variable "lambda_name" {
+  type = string
+  description = "The full name of the lambda function"
+}
+variable "lambda_subdir" {
```

Comment:
Dette er kanskje _litt_ scope creep, men noe jeg har tenkt på en stund...

Jeg må innrømme at jeg syns det er en uting at vi trenger `dir` og `subdir` for å lage en lambda med denne modulen, siden det gjør at modulen krever en spesifikk mappe-struktur. Kan vi ikke bare samle det til én path: `lambda_dir`? Da blir det kanskje behov for litt mer string concatenation de fleste plassene vi bruker det, men jeg syns det blir mye lettere å forholde seg til.

Rent praktisk så kan jeg ikke se at vi trenger "outer path" eller "common dir for lambda code" for noe som helst annet enn at det er der vi putter de zip-filene i `/out/`. Men `/out/` kan vi like godt putte _inni_ mappen til lambda-koden?

Reply:
Legg merke til at `dir` nå har en default-verdi, så trenger i praksis kun å sette `subdir` (og at navnene er... noe lettere å forstå)!
Kan forstå ønsket om å bare ha én full path, men tror heller vi tar det siden i så fall!
Kan probably ha de `out/`-filene flere steder ja, men ikke helt uvanlig å ha en "out/build/bin"-mappe heller? Lurer også litt på om vi skulle ha skrevet noe for å rydde opp de filene (etter apply?) også...

har laget et kort i stedet? https://trello.com/c/p5oIvuvM/857-terraform-cdp-lambda-lambda

Reply:
aokei da

Reply:
Resolver denne så tar vi det siden 👍